sdk/cosmos/azure-cosmos/azure/cosmos/user.py

```diff
@@ -291,10 +296,8 @@ def delete_permission(self, permission, **kwargs):
         """
         request_options = build_options(kwargs)
         response_hook = kwargs.pop('response_hook', None)
-
-        result = self.client_connection.DeletePermission(
+        self.client_connection.DeletePermission(
             permission_link=self._get_permission_link(permission), options=request_options, **kwargs
         )
-
         if response_hook:
-            response_hook(self.client_connection.last_response_headers, result)
+            response_hook(self.client_connection.last_response_headers, None)
```

Comment:
I need to do a bunch of tests here to confirm that this isn't breaking, as I have changed _all_ DELETE callbacks to use `None` directly.

Reply:
Okay I've tested `delete_database`, `delete_container`, `delete_item`, `delete_user` and `delete_permission`. All of these were fine.
I wasn't able to test `delete_items_by_partitionkey` as our sub isn't enabled for this feature - but I see this request is using `POST` rather than a `DELETE` and I couldn't find any REST docs to confirm behaviour, so I've reverted the change there.
The script methods `delete_trigger`, `delete_sproc` and `delete_udf` do not support response hook.
I wasn't able to test `delete_conflict`, nor could I find REST docs on it, but the underlying call is a `DELETE` so I'm assuming it follows the same pattern as the other delete resource operations.

Reply:
all deletes return None, even delete_all_items_by_partition_key 👍